### PR TITLE
Fix vercel deployments

### LINKED ISSUE #9: Deployment
Ask Yousef

### PR DESCRIPTION
**Pull request by mattwalo32**
## Summary

Get the frontend and backend to properly run and connect to each other.

## Changes

- Renamed variables to begin with ```REACT_APP```
- Use ```VERCEL_URL``` to reference api on a per-deployment basis
- Added .env.production file to make VERCEL_URL available at runtime

Closes #9

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "lint": "eslint --fix src -c .eslintrc.json --ext js",
     "format": "prettier --write \"./**/*.{js,jsx,json,md}\"",
     "format:check": "prettier --check \"./**/*.{js,jsx,json,md}\"",
-    "test": "jest"
+    "test": "jest --passWithNoTests"
```
Comment (alicesf2) [inline api/package.json:12]: which test was failing? i think I fixed it in my mongo connection PR

File: .gitignore [hunk 2]
```diff
@@ -69,8 +69,9 @@ typings/
 .yarn-integrity
 
 # dotenv environment variables file
-.env
-.env.test
+#*.env
+#*.env.test
+#*.env.production
```
Comment (alicesf2) [inline .gitignore:74]: I think we can keep the .env files in the .gitignore for now

**Review by alicesf2 — CHANGES_REQUESTED** [review]
Good work 👍  I just wanted to check if the test that failed was the same one I fixed in my pr, cuz if so it should work when u merge with main

File: .gitignore (hunk 2 above)
Comment (mattwalo32) [reply .gitignore]: Ok, I'll add that back

File: api/package.json (hunk 1 above)
Comment (mattwalo32) [reply api/package.json:12]: Ok, I'll get your changes in

**Review by Qrtn — CHANGES_REQUESTED** [review]
Otherwise looks good!

File: api/config/dev.env [hunk 3]
```diff
@@ -1 +1,5 @@
-MONGO_URL=dev-mongo
\ No newline at end of file
+MONGO_URL=mongodb+srv://memberdb-admin:GzUoUaNWRHfZP7Fz@cluster0.hyqeo.mongodb.net/dev?retryWrites=true&w=majority
+GOOGLE_CLIENT_ID=660704467078-j1cehuk1cr74vctddhh261am13nama98.apps.googleusercontent.com
+GOOGLE_CLIENT_SECRET=AJvH31jF850hDEYCcyKc2nan
+OAUTH_CALLBACK_URI=http://localhost:9000/api/auth/callback
+SESSION_SECRET=keyboard cat
```
[the remark this answers is not in the document]
Comment (mattwalo32) [reply api/config/dev.env:5]: o yeah, that shouldn't be there

**Review by Qrtn — APPROVED** [review]

File: api/package.json (hunk 1 above)
Comment (alicesf2) [reply api/package.json:12]: Could you remove the --passWithNoTests flag?

**Review by alicesf2 — CHANGES_REQUESTED** [review]
One small thing but otherwise looks good!

**Pull request merged**